aiutils/vis/image_io.py

```diff
@@ -1,54 +1,55 @@
-from termcolor import colored
 from PIL import Image
 import numpy as np
 import pdb
 
-
-def raise_exception(e, err_str):
-    pkg_err_str = 'Exception raised in package pyAIUtil: \n{}'.format(err_str)
-    print colored(pkg_err_str, 'red')
-    raise
-
-
-def check_assertion(cond, err_str):
-    pkg_err_str = '\nAssertion failed in package pyAIUtil: \n{}'.format(
-        err_str)
-    assert (cond), colored(pkg_err_str, 'blue')
+import aiutils.devtools.err_msg as err_msg
 
 
 def imread(filename):
+    """
+    Matlab like function for reading an image file.
+
+    Returns:
+    np_im (numpy.ndarray): h x w x 3 ndarray for color images and 
+        h x w for grayscale images
+    """
     try:
         im = Image.open(filename)
-
     except IOError as e:
         err_str = 'imread could not open file {}'.format(filename)
-        raise_exception(e, err_str)
+        err_msg.raise_exception(e, err_str)
 
     assert_cond = im.mode == 'RGB' or im.mode == 'L'
     err_str = \
         "imread only supports 'RGB' and 'L' modes, found '{}'".format(im.mode)
-    check_assertion(assert_cond, err_str)
+    err_msg.check_assertion(assert_cond, err_str)
 
     im_seq = np.array(im.getdata(), np.uint8)
 
     if im.mode == 'RGB':
         np_im = im_seq.reshape(im.size[1], im.size[0], 3)
-
     elif im.mode == 'L':
         np_im = im_seq.reshape(im.size[1], im.size[0])
 
     return np_im
 
 
 def imshow(np_im):
-    assert_cond = len(np_im.shape)==3 or len(np_im.shape)==2
+    """
+    Matlab like function for displaying a numpy ndarray as an image
+
+    Args:
+    np_im (numpy.ndarray): h x w x 3 ndarray for color images and 
+        h x w for grayscale images with pixels stored in uint8 format
+    """
+    assert_cond = len(np_im.shape) == 3 or len(np_im.shape) == 2
     err_str = 'imshow expects ndarray of dimension h x w x c (RGB) or h x w (L)'
-    check_assertion(assert_cond, err_str)
-    
+    err_msg.check_assertion(assert_cond, err_str)
+
     if len(np_im.shape) == 3:
-        check_assertion(np_im.shape[2]==3, 'imshow expected 3 channels')
+        err_msg.check_assertion(np_im.shape[2] == 3,
+                                'imshow expected 3 channels')
         im = Image.fromarray(np_im, 'RGB')
-
     else:
         im = Image.fromarray(np_im, 'L')
 
```

Comment:
It uses PIL's Image.show function. I don't think it uses matplotlib. My reason was just to keep dependencies to minimum and use PIL for every thing rather that getting confused between matplotlib and PIL in different parts of the package.